### PR TITLE
Pullquote attempt 3

### LINKED ISSUE #14: Pull quote plugin
#### Feature description:

Simple plugin for the CMS editor that allows editorial to add a pullquote
#### Proposed solution:
- Have  button in editor that opens a dialog box that requires four pieces of information from the user (quote, author image, author name, author title/position)
- Plugin will insert HTML structure of a pullquote (already created) with styling that Natalie used for intern project.
- Image will be a static link for right now (so I don't have to go into the rabbit hole of integrating with the diveimage portion of the CMS.
#### This issue can be closed when:

Editor/tester can insert a pull quote with no confusion. No errors are caused by inserted HTML and the elements are styled correctly.
#### Priority level / Rationale:

Medium, I think I can finish it quickly and it would be useful to editorial staff if it is working correctly.

### PR DESCRIPTION
**Pull request by haydenwagner**
### PR Cherry Picking Craziness

First pr was this one by me: https://github.com/industrydive/django-ckeditor/pull/18 , but I had 88 files editor because of some spacing edits, so Miriam made a cleaned up PR: https://github.com/industrydive/django-ckeditor/pull/21 with only the actual changed files from my first PR. 

She wanted me to be involved in the PR if I wanted, so I made a few edits to those files and then remade her PR here

---
#### Which issue(s) does this PR address?

closes #14  (named 2050 to match up with the divesite issue number: industrydive/divesite#2050)
#### What does this PR do?

Pulls in a new directory in the ckeditor plugin dir, which makes a pull quote plugin for editors to use. It also makes minor edits to the diveimage plugin so they play nice.
#### Where should the reviewer start?

**Make sure the equivalent divesite PR (https://github.com/industrydive/divesite/pull/2162) has been resolved if it can be, or this PR you are reviewing wont work (needs some changes to the settings.py file that the divesite pr provides)**

Go make a new newspost, and scroll down to the editor and look for the icon that looks like 
a speech bubble (quote .etc), and click it. 

This should open up a dialog in which you can provide a quote, image url, speaker name, and speaker title to create a pullquote.
#### How should this be manually tested?

Create that SOB and then view the newspost and see how she looks. It should look like a well-styled pullquote (see screenshots below)
- Test it in desktop,
- Test it in mobile,
- Open the editor back up and make another pullquote
- add some text to this editor
- rearrange the pull quotes in the editors
- edit the text and image in the pullquote
- leave an image field blank and close the dialog and see if it works (no image in the pullquote in the editor)
- save the page 
- view the actual newspost again and make sure everything looks peachy
#### Screenshots
- good desktop pullquote

### CODE REVIEW

**Comment by rschuster3** [comment]
@MiriamSexton Can we get to this today? I'd love to take another look at the Divesite PR and merge it in, but I need this one to be merged in first.

**Comment by MiriamSexton** [comment]
I'm not comfortable merging this in while our solution for a non existent image is `src=" "` and adding class that with `display:none`. I also have a few other quibbles, including that you can insert into the middle of a paragraph thus making the widget break the paragraph in two, and the hack into `diveimage` but I can get over those.

**Comment by rschuster3** [comment]
What's wrong with that? It's not the most elegant solution, but it does solve the problem while not (I think) not creating a new one. My worry is if this doesn't get merged in today based on a technicality, it won't be merged in at all for months.

File: ckeditor/static/ckeditor/ckeditor/plugins/divepullquote/dialogs/divepullquote.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+							this.setValue(widget.data.img_src);
+						},
+						commit: function( widget ) {
+							//get the actual img dom element, not some ckeditor psuedo element wrapped bullshit
+							var img = widget.editables.imgDiv.getFirst().$;
+
+							//checks for an empty value or one space...we allow user to create pullquote without an image,
+							//but we need to change the pullquote class styling to not display on the page
+							if(this.getValue() === '' || this.getValue() === ' '){
+								widget.setData('img_src', this.getValue());
```
Comment (elidickinson) [inline ckeditor/static/ckeditor/ckeditor/plugins/divepullquote/dialogs/divepullquote.js]: Can we not just remove the img node here if it isn't being used? Something like `img.parentNode.removeChild(img)`

**Comment by elidickinson** [comment]
Agree with Miriam that it's icky to put it live with broken HTML. Ideally the PR is adjusted to remove the img rather than hide it. Or we put it live, but img src is a required field. Or it defaults to a valid link to a placeholder image (even a 1x1 blank.gif).

**Comment by rschuster3** [comment]
I added code to not only hide the img attribute if no img link is provided, but to fill its "src" with a link to our placeholder img. So even though img is hidden, its src is not empty.

I chose to not just remove img since we'd need to do some ugly JS to add it back in to the right place manually every time someone decided to not have an img, then changed their mind and decided they *do* want to have an img. Hiding it and assigning a placeholder is much easier/more explicit in the code.

@MiriamSexton Feel free to now QA this particular plugin on the industrydive/divesite branch 2050-ckeditor-pullquote-plugin-2. If you're happy, we can pull this PR in first, then update the divesite branch to use the relevant commit and pull it in as well.

**Comment by MiriamSexton** [comment]
The placeholder image doesn't show up for me: I get a 403 error: `https://d12v9rtnomnebu.cloudfront.net/dive_static/diveimages/corporate_site/teampage/square_profiles/placeholder-200.png`

**Comment by MiriamSexton** [comment]
Given that it's hidden, I don't know how much that matters.

Let's get this sucker live.

**Pull request merged**